src/writer.rs

```diff
@@ -161,11 +162,11 @@ impl<D: Distance> Writer<D> {
         loop {
             match n_trees {
                 Some(n_trees) if thread_roots.len() >= n_trees => break,
-                None if self.next_tree_id as usize >= self.n_items => break,
+                None if self.next_tree_id >= (self.n_items - 1) as u32 => break,
                 _ => (),
             }
 
-            let tree_root_id = self.make_tree(wtxn, &item_indices, true, &mut rng)?;
+            let tree_root_id = self.make_tree(wtxn, item_indices.clone(), true, &mut rng)?;
```

Comment:
Yes, but I just found out it was faster this way since it avoids cloning once per dummy node.
And since we know we have more than one dummy node per tree, it's worth it overall.